pulp-glue/pulp_glue/common/context.py

```diff
@@ -280,10 +280,12 @@ def call(
         return result
 
     @classmethod
-    def _check_task_finished(cls, task: EntityDefinition) -> bool:
+    def _check_task_finished(cls, task: EntityDefinition, expect_cancel: bool = False) -> bool:
         task_href = task["pulp_href"]
 
         if task["state"] == "completed":
+            if expect_cancel:
+                raise PulpException(_("The task {task_href} meant to be canceled, completed instead."))
             return True
         elif task["state"] == "failed":
```

Comment:
not sure, but do you want to check here for the `expect_cancel` too?

Reply:
I think no. If i cancel and it fails instead, i want to know that.